backend/src/main/java/pl/seahawks/vacationes/providers/ItakaProvider.java

```diff
@@ -0,0 +1,109 @@
+package pl.seahawks.vacationes.providers;
+
+import com.gargoylesoftware.htmlunit.BrowserVersion;
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.html.*;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.springframework.context.annotation.Configuration;
+import pl.seahawks.vacationes.dto.OfferDto;
+import pl.seahawks.vacationes.dto.OfferRequirementsDto;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Configuration
+public class ItakaProvider implements Provider {
+
+  private final String BASE_URL_ITAKA = "https://www.itaka.pl";
+  private final String BASE_URL_THE_BEST_OFFER = "https://www.itaka.pl/wyniki-wyszukiwania/wakacje/?review-rate=50";
+  private final String BASE_SEARCH_URL_WITH_PARAMS =
+      "https://www.itaka.pl/wyniki-wyszukiwania/wakacje/?date-from=%s&date-to=%s&adults=%s&dest-region=%s";
+
+  private final WebClient webClient;
+
+  public ItakaProvider() {
+    webClient = new WebClient(BrowserVersion.CHROME);
+    webClient.getOptions().setJavaScriptEnabled(false);
+    webClient.getOptions().setCssEnabled(true);
+    webClient.getOptions().setThrowExceptionOnScriptError(true);
+    webClient.waitForBackgroundJavaScript(100000);
+    webClient.setJavaScriptTimeout(100000);
+    webClient.waitForBackgroundJavaScript(100000);
+  }
+
+  @Override
+  public List<OfferDto> getOfferList(OfferRequirementsDto offerRequirementsDto) throws IOException {
+    String url =
+        String.format(
+            BASE_SEARCH_URL_WITH_PARAMS,
+            offerRequirementsDto.getDateFrom(),
+            offerRequirementsDto.getDateTo(),
+            offerRequirementsDto.getAdults(),
+            offerRequirementsDto.getRegion());
+    Document webWithGeneralOffers = Jsoup.connect(url).get();
+    Elements articles = webWithGeneralOffers.getElementsByTag("article");
+    List<OfferDto> listItakaGeneralOferr = new ArrayList<>();
+    articles.forEach(
```

Comment:
tu masz duplikacje kodu. od lini 51 do 59 masz to samo co 67 do 74, warto wynieść do osobnej metody.